src/servers/src/postgres/handler.rs

```diff
@@ -194,35 +357,52 @@ impl ExtendedQueryHandler for PostgresServerHandler {
     async fn do_query<C>(
         &self,
         _client: &mut C,
-        _portal: &Portal<Self::Statement>,
+        portal: &Portal<Self::Statement>,
         _max_rows: usize,
     ) -> PgWireResult<Response>
     where
         C: ClientInfo + Unpin + Send + Sync,
     {
-        Ok(Response::Error(Box::new(ErrorInfo::new(
-            "ERROR".to_owned(),
-            "XX000".to_owned(),
-            "Extended query is not implemented on this server yet".to_owned(),
-        ))))
+        let (_, sql) = portal.statement().statement();
+
+        // manually replace variables in prepared statement
+        let mut sql = sql.to_owned();
+        for i in 0..portal.parameter_len() {
+            sql = sql.replace(&format!("${}", i + 1), &parameter_to_string(portal, i)?);
```

Comment:
seems simply replacing strings is vulnerable to sql injection?

Reply:
This is part is why this implementation is a proof-of-concept. Text data is quoted in `parameter_to_string` so I think it's relatively safe.